### ISSUE TITLE
`Naming/PredicateMethod` false positive with ternary returning strings

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#14262]
## Description

```ruby
module CustomDecorator
  def human_allowed?
    inverse ? 'no' : 'yes'
  end
end
```

## Rubocop

```console
1.76.1 (using Parser 3.3.8.0, rubocop-ast 1.45.1, analyzing as Ruby 2.7, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.22.1
  - rubocop-factory_bot 2.27.1
  - rubocop-migration 0.7.1
  - rubocop-performance 1.25.0
  - rubocop-rails 2.32.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.6.0
  - rubocop-rspec_rails 2.31.0
  - rubocop-thread_safety 0.7.2
```

**Comment by koic** [comment]
Since the return value is a string, not a boolean, this appears to be the expected behavior.

```console
$ cat example.rb
module CustomDecorator
  def human_allowed?
    inverse ? 'no' : 'yes'
  end
end

$ bundle exec rubocop --only Naming/PredicateMethod
Inspecting 1 file
C

Offenses:

example.rb:2:7: C: Naming/PredicateMethod: Non-predicate method names should not end with ?.
  def human_allowed?
      ^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

**Comment by bbatsov** [comment]
Indeed. This should not be named as a predicate method, so the cop discovered a real problem with the code.
